ci/configure_cpp_static.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+rapids-configure-conda-channels
+
+source rapids-date-string
+
+rapids-logger "Configure static cpp build"
+
+ENV_YAML_DIR="$(mktemp -d)"
+REQUIREMENTS_FILE="${ENV_YAML_DIR}/requirements.txt"
+
+rapids-dependency-file-generator \
+  --output requirements \
+  --file_key test_static_build \
+  --matrix "" | tee "${REQUIREMENTS_FILE}"
+
+python -m pip install -r "${REQUIREMENTS_FILE}"
+pyenv rehash
+
+cmake -S cpp -B build_static -GNinja -DBUILD_SHARED_LIBS=OFF -DBUILD_TESTS=OFF
```

Comment:
The tests will not currently configure this way because of how the test util and the stream identification are set up. I don't want to try to fix that in this PR.

Reply:
That's fine. Resolving.